File: Upcoder/Lập trình cơ bản/2.Lệnh rẽ nhánh/Phương trình bậc 2/MINIGAME16.1PTBAC2.cs
//Code này thì chỉ được 2.67đ
using System;

namespace GiaiPhuongTrinh
{
    public class Program
    {
        public static void Main(string[] args)
        {
            double a, b, c, d, x1, x2;
            string[] input = Console.ReadLine().Split(' ');
            a = double.Parse(input[0]);
            b = double.Parse(input[1]);
            c = double.Parse(input[2]);

            if (a == 0 && b == 0 && c == 0)
            {
                Console.WriteLine(-1);
                return;
            }

            if (a == 0 && b == 0 && c != 0)
            {
                Console.WriteLine(0);
                return;
            }

            if (a == 0 && b != 0)
            {
                Console.WriteLine(1);
                x1 = c / -b;
                Console.WriteLine($"{x1:F10}");
                return;
            }

            if (a != 0)
            {
                d = b * b - 4 * a * c;

                if (d < 0)
                {
                    Console.WriteLine(0);
                    return;
                }

                if (d == 0)
                {
                    Console.WriteLine(1);
                    x1 = -b / (2 * a);
                    Console.WriteLine($"{x1:F10}");
                    return;
                }

                if (d > 0)
                {
                    Console.WriteLine(2);
                    x1 = (-b - Math.Sqrt(d)) / (2 * a);
                    x2 = (-b + Math.Sqrt(d)) / (2 * a);

                    // Sắp xếp x1 và x2
                    if (x1 > x2)
                    {
                        double temp = x1;
                        x1 = x2;
                        x2 = temp;
                    }
                    Console.WriteLine($"{x1:F10}");
                    Console.Write($"{x2:F10}");
                }
            }
        }
    }
}

File: Upcoder/Lập trình cơ bản/2.Lệnh rẽ nhánh/Phương trình bậc 2/MINIGAME16.1PTBAC2.py
#đúng test case mẫu mà không được điểm
a,b,c = list(map(int,input().split()))

if a == 0:
    if b == 0:
        if c == 0:
            print(-1,end='')
        else: print(0,end='')
    else:
        print(1)
        x = -c/b
        if abs(x) == 0: x = 0
        print('%.10f' %x,end='')
else:
        d = b**2 - 4*a*c
        if d < 0: print(0,end='')
        elif d==0:
            print(1)
            x = (-b/(2*a))
            if abs(x) == 0: x = 0
            print('%.10f' %x,end='')
        else:
            print(2)
            x1 = ((-b-(d**0.5))/(2*a))
            x2 = ((-b+(d**0.5))/(2*a))
            
            if abs(x1) == 0: x1 = 0
            if abs(x2) == 0: x2 = 0
            if x1 > x2:
                tmp = x1
                x1 = x2
                x2 = tmp
            
            print('%.10f' %x1)
            print('%.10f' %x2 ,end = '')